lib/libstorj/src/storj.c

```diff
@@ -0,0 +1,782 @@
+    req->response = json_object_new_object();
+
+    free_bucket_list(&bucket_list);
+}
+
+static void get_bucket_request_worker(uv_work_t *work)
+{
+    get_bucket_request_t *req = work->data;
+
+    BucketInfo bucket_info = get_bucket_info(req->project_ref,
+                                             strdup(req->bucket_name),
+                                             STORJ_LAST_ERROR);
+    STORJ_RETURN_SET_REQ_ERROR_IF_LAST_ERROR
+
+    req->bucket = malloc(sizeof(storj_bucket_meta_t));
+
+    char created_str[32];
+    time_t created_time = (time_t)bucket_info.created;
+    strftime(created_str, 32, "%DT%T%Z", localtime(&created_time));
+
+    req->bucket->name = strdup(bucket_info.name);
+    req->bucket->id = strdup(bucket_info.name);
+    req->bucket->created = strdup(created_str);
+    req->bucket->decrypted = true;
+    // NB: this field is unused; it only exists for backwards compatibility as it is
+    //  passed to `json_object_put` by api consumers.
+    //  (see: https://svn.filezilla-project.org/svn/FileZilla3/trunk/src/storj/fzstorj.cpp)
+    req->response = json_object_new_object();
+
+    free_bucket_info((BucketInfo *)&bucket_info);
+}
+
+static void delete_bucket_request_worker(uv_work_t *work)
+{
+    delete_bucket_request_t *req = work->data;
+
+    delete_bucket(req->project_ref, strdup(req->bucket_name), STORJ_LAST_ERROR);
+    STORJ_RETURN_SET_REQ_ERROR_IF_LAST_ERROR
+
+    // NB: http "no content" success status code.
+    req->status_code = 204;
+}
+
+static void list_files_request_worker(uv_work_t *work)
+{
+    list_files_request_t *req = work->data;
+
+    BucketRef bucket_ref = open_bucket(req->project_ref, strdup(req->bucket_id),
+                                       strdup(req->encryption_access),
+                                       STORJ_LAST_ERROR);
+    STORJ_RETURN_SET_REQ_ERROR_IF_LAST_ERROR;
+
+    ObjectList object_list = list_objects(bucket_ref, NULL, STORJ_LAST_ERROR);
+    STORJ_RETURN_SET_REQ_ERROR_IF_LAST_ERROR;
+
+    req->total_files = object_list.length;
+
+    if (object_list.length > 0) {
+        req->files = malloc(sizeof(storj_file_meta_t) * object_list.length);
+
+        ObjectInfo object_item;
+        for (int i = 0; i < object_list.length; i++) {
+            object_item = object_list.items[i];
+            storj_file_meta_t *file = &req->files[i];
+
+            char created_str[32];
+            time_t created_time = (time_t)object_item.created;
+            strftime(created_str, 32, "%DT%T%Z", localtime(&created_time));
+
+            file->created = strdup(created_str);
+            file->mimetype = strdup(object_item.content_type);
+            file->id = strdup(object_item.path);
+            file->bucket_id = strdup(object_item.bucket.name);
+            file->filename = strdup(object_item.path);
+            file->decrypted = true;
+
+            // TODO: if we want to populate size we need to
+            //  get object meta for each file.
```

Comment:
Why is this? In the Java bindings, we have a `Size` field in the `ObjectInfo` class. I don't see no reason to be the same for the C bindings. Perhaps, it was just missed to be added.